mlir/lib/Transforms/RemoveDeadValues.cpp

```diff
@@ -178,6 +181,17 @@ static void cleanSimpleOp(Operation *op, RunLivenessAnalysis &la) {
   if (!isMemoryEffectFree(op) || hasLive(op->getResults(), la))
```

Comment:
I never looked into the liveness analysis, but shouldn't a value be considered "alive" it is passed as a function call operand?

Reply:
From what I understand, if the parameter is not used inside the function, the value (argument) will not be considered "alive".